packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -157,6 +157,8 @@ class FlutterPlugin implements Plugin<Project> {
     }
 
     private void addFlutterJarProvidedDependency(Project project) {
+        if (project.state.failure)
+            return
```

Comment:
I'd suggest having the `return` on the same line as the `if`, or add `{}`'s around `return`, to make it clear they belong together.

Especially since semi-colons are optional in Groovy, so new line usually means new statement.